src/misc/ssh/ssh_decode_sequence_multi.c

```diff
@@ -0,0 +1,136 @@
+/* LibTomCrypt, modular cryptographic library -- Tom St Denis
+ *
+ * LibTomCrypt is a library that provides various cryptographic
+ * algorithms in a highly modular and flexible manner.
+ *
+ * The library is free for all purposes without any express
+ * guarantee it works.
+ */
+#include "tomcrypt_private.h"
+#include <stdarg.h>
+
+/**
+   @file ssh_decode_sequence_multi.c
+   SSH data type representation as per RFC4251, Russ Williams
+*/
+
+#ifdef LTC_SSH
+
+/**
+  Decode a SSH sequence using a VA list
+  @param out    [out] Destination for data
+  @param outlen [in/out] Length of buffer and resulting length of output
+  @remark <...> is of the form <type, data> (int, void*) except for string <type, data, size>
+  @return CRYPT_OK on success
+*/
+int ssh_decode_sequence_multi(const unsigned char *in, unsigned long inlen, ...)
+{
+   int           err;
+   ssh_data_type type;
+   void          *data;
+   va_list       args;
+   unsigned long size, bufsize;
+
+   LTC_ARGCHK(in    != NULL);
+
+   /* Decode values from buffer */
+   va_start(args, inlen);
+   for (;;) {
+      type = (ssh_data_type)va_arg(args, int);
+      data = va_arg(args, void*);
+      LTC_UNUSED_PARAM(data);
+
+      if (type == LTC_SSHDATA_EOL) {
+         break;
+      }
+
+      switch (type) {
+         case LTC_SSHDATA_BYTE:
+            size = ((unsigned long)in[0]);
+            in++;
+            *(unsigned long *)data = size;
+            break;
+         case LTC_SSHDATA_BOOLEAN:
+            size = ((unsigned long)in[0]);
+            in++;
+            /*
+               The value 0 represents FALSE, and the value 1 represents TRUE.  All non-zero values MUST be
+               interpreted as TRUE; however, applications MUST NOT store values other than 0 and 1.
+            */
+            *(unsigned long *)data = (size)?1:0;
+            break;
+         case LTC_SSHDATA_UINT32:
+            size = ((unsigned long)in[0] << 24) |
+                   ((unsigned long)in[1] << 16) |
+                   ((unsigned long)in[2] << 8) |
+                   ((unsigned long)in[3]);
+            in += 4;
+            *(unsigned long *)data = size;
+            break;
+         case LTC_SSHDATA_UINT64:
+            size = ((unsigned long)in[0] << 56) |
```

Comment:
I doubt this works on all architectures [\*cough\*](https://github.com/libtom/libtomcrypt/issues/306#issuecomment-333765095) ;)

Reply:
Btw. we also have types for exactly that purpose, `ulong32` and `ulong64` (see `tomcrypt_cfg.h` for details)

Reply:
Ah, those were exactly what I was looking for. Cheers! Fixing now :)